source/require-one.d.ts

```diff
@@ -0,0 +1,30 @@
+/**
+Create a type that requires exactly one of the given properties, but not more. The remaining properties are kept as is.
+
+Use cases:
+- Creating interfaces for components that only need one of the properties to display properly.
+- Declaring generic properties in a single place for a single use case that get narrowed down via `RequireOne`.
+
+The caveat with `RequireOne` is that TypeScript doesn't always know at compile time every property that will exist at runtime. Therefore `RequireOne` can't do anything to prevent extra properties it doesn't know about.
+
```

Comment:
Has there been any issues on the TypeScript issue tracker about a type like this? If so, would be good to include references.

Reply:
I agree, couldn't find the myself though.